ui/src/store/resource.test.ts

```diff
@@ -607,4 +609,42 @@ describe('Store functions', () => {
       }
     );
   });
+
+  it('makes sure to add resources with same name but from different catalog', (done) => {
+    const store = ResourceStore.create(
+      {},
+      {
+        api,
+        categories: CategoryStore.create({}, { api })
+      }
+    );
+
+    expect(store.isLoading).toBe(true);
+
+    when(
+      () => !store.isLoading,
+      () => {
+        const item = Resource.create({
+          id: 44,
+          name: 'golang-build',
+          uniqueKey: 'tekton-hub/Task/golang-build',
+          catalog: 2,
+          kind: 'Task',
+          latestVersion: 47,
+          displayVersion: 47,
+          tags: [1],
+          rating: 5,
+          versions: [47],
+          displayName: 'golang build'
+        });
+
+        store.add(item);
+        expect(store.resources.size).toBe(8);
+
```

Comment:
can we check catalogs size here as well ?

Reply:
To add a catalog size we'll have to add a catalog first as well before